### PR TITLE
enable connection types by default

### PR DESCRIPTION
**Pull request by christianvogt**
https://issues.redhat.com/browse/RHOAIENG-14748

## How Has This Been Tested?

- apply the CRD changes to your test cluster, thus removing the `disableConnectionTypes` feature flag
- if testing locally, run a local backend server with the changes
- observe the UI shows connection types enabled by default

## Test Impact

N/A

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

cc @andrewballantyne @dgutride @emilys314

### CODE REVIEW

File: docs/dashboard-config.md [hunk 1]
```diff
@@ -30,14 +30,14 @@ The following are a list of features that are supported, along with there defaul
...
 | disableDistributedWorkloads  | false   | Disables Distributed Workload Metrics from the dashboard.                                            |
 | disableModelRegistry         | false   | Disables Model Registry from the dashboard.                                                          |
-| disableServingRuntimeParams  | false    | Disables Serving Runtime params from the dashboard.                                                  |
-| disableConnectionTypes       | true    | Disables creating custom data connection types from the dashboard.                                   |
-| disableStorageClasses        | false    | Disables storage classes settings nav item from the dashboard.                                       |
-| disableNIMModelServing       | true    | Disables components of NIM Model UI from the dashboard.   
+| disableServingRuntimeParams  | false   | Disables Serving Runtime params from the dashboard.                                                  |
+| disableConnectionTypes       | false   | Disables connection types from the dashboard.                                                        |
```
Comment (andrewballantyne) [inline docs/dashboard-config.md:38]: Can you clarify this is not a valid property for the CR but is under the hood? I feel it's odd to have it here if it's not clarified more.

Comment (christianvogt) [reply docs/dashboard-config.md:38]: @andrewballantyne sorry that should have been removed after I changed my mind to completely remove it from the CRD. Let me update

**Review by andrewballantyne — APPROVED** [review]
Whenever you get PM approval

**Pull request merged**